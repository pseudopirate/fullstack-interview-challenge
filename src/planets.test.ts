import fs from 'fs';
import path from 'path';
import {
    Destination, listAvailableDestinations, listDestinations, listPlanets,
} from './planets';

describe('planets', () => {
    describe('listPlanets()', () => {
        it('should return planets list', async () => {
            const planets = await listPlanets();

            expect(Array.isArray(planets)).toBeTruthy();
            expect(
                planets.every((planet) => Boolean(planet.name) && Boolean(planet.code)),
            ).toBeTruthy();
        });
    });

    describe('listDestinations()', () => {
        it('should return destinations list', async () => {
            const dests = await listDestinations();

            expect(Array.isArray(dests)).toBeTruthy();
            // expect(
            //     dests.every((dest) => ('name' in planet)),
            // ).toBeTruthy();
        });
    });

    jest.mock('../planets', () => ({
        listDestinations: async () => {
            const filepath = path.join(__dirname, '../dataset.test.json');
            const planets = await fs.promises.readFile(filepath, 'utf8');

            return JSON.parse(planets) as Destination[];
        },
    }));

    describe('listAvailableDestinations(date: number)', () => {
        it('should return empty list', async () => {
            const date = new Date('2222-11-11').getTime();
            const dests = await listAvailableDestinations(date);
            expect(dests).toEqual([]);
        });

        it('should match snapshot', async () => {
            const date = new Date('2022-11-06').getTime();
            const dests = await listAvailableDestinations(date);
            expect(dests).toMatchSnapshot();
        });
    });
});
