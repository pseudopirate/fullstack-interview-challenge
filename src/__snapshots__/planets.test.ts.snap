// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`planets listAvailableDestinations(date: number) should match snapshot 1`] = `
Array [
  Object {
    "availability": 6,
    "data": "2022-11-06",
    "destination": "CUS",
    "origin": "NAB",
    "price": 249.55,
  },
  Object {
    "availability": 5,
    "data": "2022-11-06",
    "destination": "HOH",
    "origin": "NAB",
    "price": 425.68,
  },
  Object {
    "availability": 8,
    "data": "2022-11-06",
    "destination": "HOH",
    "origin": "TAT",
    "price": 159.78,
  },
  Object {
    "availability": 4,
    "data": "2022-11-06",
    "destination": "NAB",
    "origin": "CUS",
    "price": 316.47,
  },
  Object {
    "availability": 6,
    "data": "2022-11-06",
    "destination": "NAB",
    "origin": "HOH",
    "price": 469.6,
  },
  Object {
    "availability": 7,
    "data": "2022-11-06",
    "destination": "TAT",
    "origin": "HOH",
    "price": 321.23,
  },
]
`;
